packages/pwa-kit-extension-sdk/src/react/classes/ApplicationExtension.ts

```diff
@@ -73,4 +96,84 @@ export class ApplicationExtension<
     public beforeRouteMatch(routes: RouteProps[]): RouteProps[] {
         return routes
     }
+
+    /**
+     * Called on the server to serialize the extension data that will be sent to the client.
+     *
+     * @returns SerializedExtension - The serialized extension data.
+     * @throws Error if the routes have not been loaded.
+     */
+    public serialize(): SerializedExtension {
+        if (this._cachedRoutes === null) {
+            throw new Error('Routes have not been loaded. Call getRoutes() before serializing')
+        }
+        const serializedRoutes = this._cachedRoutes.map((route) => {
+            if (!route.component?.displayName) {
+                throw new Error(
+                    `Component for route with path "${
+                        route.path
+                    }" is missing a displayName in ${this.getName()} extension`
+                )
+            }
+            return {
+                path: route.path,
+                componentName: route.component?.displayName
+            }
```

Comment:
I think it's better if have all the route's properties defined here in `serialize`. So for example, we don't need to worry about adding `exact` over in the the corresponding `deserialize` method.

```suggestion
            return {
                path: route.path,
                componentName: route.component?.displayName,
                exact: true
            }
```

Reply:
very good point!